README.md

```diff
@@ -1,63 +1,161 @@
 # npl-integrations
 
-## Pre-requisites
-Make sure your `hosts` file includes keycloak at 127.0.0.1:
+The npl-integrations repo contains a sample project that demonstrates how to integrate with [Noumena's Operating Engine](https://documentation.noumenadigital.com/engine/applications/engine/) using
+different programming languages and frameworks.
+It includes a Python listener service, a typescript React web app, and a Python Streamlit UI.
 
-![img.png](docs/img.png)
+The underlying NPL code is an extension of the IOU, which can be found
+in the [npl-starter](https://github.com/NoumenaDigital/npl-starter) repository.
 
 ## Building and running with docker locally
-1. Run `mvn clean install` to build and generate NPL-api and clients.
-2. Run `docker compose up --build -d` to ensure python and frontend are also build, then create and start containers.
 
-URLs:
-* Swagger UI of Engine APIs: http://localhost:12000/swagger-ui/
-* Keycloak admin console: http://localhost:11000
-* Webapp: http://localhost:8090
+The project can be built and run locally using docker-compose.
+To run the Noumena Engine, the corresponding docker image is required, which is available under license. 
+Please contact Noumena Digital for more information at [info@noumenadigital.com](mailto:info@noumenadigital.com)
+
+The npl-integrations project can be locally built and all containers run locally using docker-compose.
+
+### Pre-requisites
+
+For OS X and linux systems, make sure the `/etc/hosts` file includes the line `127.0.0.1 keycloak`
+
+![img.png](docs/img.png)
+
+If the line is not present, follow the steps below to add it by running `echo "127.0.0.1 keycloak" | sudo tee -a /etc/hosts` in the terminal.
+
+### Build & run
+
+To build and run the project, follow the steps below:
+
+1. Run `mvn clean install` to build and generate NPL-API and clients.
+2. Run `docker compose up --build -d` to ensure the Python listener service, the Python Streamlit UI and the Typescript React frontend are built in addition to the engine and its dependencies, then create and start the containers.
```

Comment:
```suggestion
2. Run `docker compose up --build -d` to ensure the Python listener service, the Python Streamlit UI and the Typescript React frontend are built in addition to the Engine and its dependencies, then create and start the containers.
```

Double check the proper nouns :D -- (sorry, this really is a nit, though Engine is probably one of the ones that actually do matter, a little bit)